tests/filecheck/transforms/convert_memref_args_to_ptr.mlir

```diff
@@ -0,0 +1,42 @@
+// RUN: xdsl-opt -p convert-memref-to-ptr{lower_func=true} --split-input-file --verify-diagnostics %s | filecheck %s
+
```

Comment:
the `--split-input-file` allows to split each case separated by

```
// -----
```

into a separate file - useful to isolate test behaviour. So, I'd use it in between the `func.func` definitions below.